lib/puppetserver/ca/utils/file_system.rb

```diff
@@ -6,6 +6,15 @@ module Ca
     module Utils
       class FileSystem
 
+        DIR_MODES = {
+          :ssldir => 0771,
+          :cadir => 0755,
+          :certdir => 0755,
+          :privatekeydir => 0750,
+          :publickeydir => 0755,
+          :signeddir => 0755
+        }
+
```

Comment:
I've added this so that our directories are now created with the correct permissions (as found in the puppet defaults.rb file)